specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1366 @@
+    /// resolve to different packages for different users on a system.
+    ///
+    /// This adds the resulting package to the process' package graph.
+    /// A process' package graph is used to search for DLLs (per Dynamic-Link Library Search Order),
+    /// WinRT objects and other resources; the caller can now load DLLs, activate
+    /// WinRT objects and use other resources from the framework package until
+    /// PackageDependencyContext.Remove() is called (or the process ends).
+    /// The package dependency Id must match a package dependency defined
+    /// for the calling user or the system (via PinForSystem) or an exception is raised.
+    ///
+    /// Each successful call adds the resolve packaged to the
+    /// calling process' package graph, even if already present. There is no
+    /// duplicate 'detection' or 'filtering' applied by the API (multiple
+    /// references to a package is not harmful). Once resolution is complete
+    /// the package stays resolved for that user until the last reference across
+    /// all processes for that user is removed via PackageDependencyContext.Remove()
+    /// (or process termination).
+    ///
+    /// This adds the resolved package to the caller's package graph, per rank.
+    /// A process' package graph is a list of packages sorted by rank in ascending
+    /// order (-infinity...0...+infinity). If package(s) are present in the
+    /// package graph with the same rankrank the resolved package is
+    /// (by default) added after others of the same rank. To add a package
+    /// before others of the same rank, specify PackageDependency.PrependIfRankCollision.
+    ///
+    /// Calls to Add() can be balanced by a PackageDependencyContext.Remove()
+    /// to remove the entry from the package graph.
+    PackageDependencyContext Add(AddPackageDependencyOptions options);
+}
+
+/// A unique identifier for a resolved package dependency
+struct PackageDependencyContextId
+{
+    UInt64 Id;
+};
+
+///TBD
+runtimeclass PackageDependencyContext : ICloseable
+{
+    /// Create an intstance of the package dependency context identified by context
+    PackageDependencyContext(PackageDependencyContextId);
+
+    /// Returns the package dependency context
+    PackageDependencyContextId Context { get; }
+
+    /// Returns the package full name of the resolved package for this context
+    String PackageFullName { get; }
+
+    /// Return true if packageDependencyContext1 and packageDependencyContext2
+    /// are associated with the same resolved package.
+    static Boolean AreEquivalent(
+        PackageDependencyContextId packageDependencyContextId1,
+        PackageDependencyContextId packageDependencyContextId2);
+
+    /// Return true if this and otherPackageDependencyContext
+    /// are associated with the same resolved package.
+    Boolean AreEquivalent(PackageDependencyContext otherPackageDependencyContext);
```

Comment:
Grammar - method name `IsEquivalent`  "foo is equivalent to bar" (compared to the prior correct `AreEquivalent`, "are foo and bar equivalent")